lib/plug/parsers.ex

```diff
@@ -236,4 +241,9 @@ defmodule Plug.Parsers do
       raise UnsupportedMediaTypeError, media_type: "#{type}/#{subtype}"
     end
   end
+
+  defp merge_path_params(%{path_params: %Plug.Conn.Unfetched{}} = conn), do: conn
```

Comment:
_If_ I understand what this does correctly, this makes sure that the existing `path_params` overwrite any params that have been overwritten by `body_params` previously, right?

Wouldn't it be nicer to have the individual parsers just update `body_params` and then to merge `params` only once, in exactly order we want, instead of "counteracting" the change the individual parsers might have done?

My approach would have been to write a function that's basically this:

```elixir
def merge_params(conn) do
%{conn | params: Map.merge(conn.query_params, Map.merge(conn.body_params, conn.path_params))}
end
```

And call it in `Plug.Conn.fetch_query_params`, `Plug.Parsers` and `Plug.Router`.